gcc/config/riscv/riscv.c

```diff
@@ -3609,12 +3630,7 @@ riscv_hard_regno_mode_ok_p (unsigned int regno, enum machine_mode mode)
     }
 
   if (FP_REG_P (regno))
-    {
-      if (mclass == MODE_FLOAT
-	  || mclass == MODE_COMPLEX_FLOAT
-	  || mclass == MODE_VECTOR_FLOAT)
-	return size <= UNITS_PER_HWFPVALUE;
-    }
+    return true;
```

Comment:
Floating point register can hold any type of value in theory if target is not force doing normalize by hardware, aarch64 allow floating point register hold any value too[1].

[1] https://github.com/riscv/riscv-gcc/blob/riscv-gcc-6.1.0/gcc/config/aarch64/aarch64.c:859
  - aarch64 has one more checking with hard_regno_nregs, but it's will check automatic by register allocator/IRA, so I think it's unneeded.